06-express-and-layered-architecture/src/cart/cart.controller.ts

```diff
@@ -0,0 +1,74 @@
+import express from "express";
+import * as cartService from "./cart.service";
+import * as orderService from "./order/order.service";
+import joi from "joi";
+import { calculateTotal } from "./middleware/cart.middleware";
+import { checkIfUserExists } from "./middleware/auth.middleware";
+
+const cartUpdateSchema = joi.object({
+  productId: joi.string().required(),
+  count: joi.number().required(),
+});
+
+const router = express.Router();
+router.use("/api/profile/cart", checkIfUserExists);
+
+router.get("/api/profile/cart", async (req, res) => {
+  const userId = res.locals.user;
+  const total = res.locals.total;
+
+  try {
+    const cart = await cartService.getOrCreateCart(userId as string);
+    res.json({ cart, total });
+  } catch (err) {
+    res.status(500).send((err as Error).message);
```

Comment:
here you could just call next(err) and then it would propagate the error to the built in error handler: https://expressjs.com/en/guide/error-handling.html. search for 'next(err)' in this document to see how it works. But basically if **any parameter is passed to the next function, Express will treat it as an error and pass it to the built in error handler**